services/horizon/internal/sse_test.go

```diff
@@ -0,0 +1,194 @@
+package horizon
+
+import (
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/kinecosystem/go/network"
+	"github.com/kinecosystem/go/services/horizon/internal/ingest"
+	"github.com/kinecosystem/go/services/horizon/internal/ledger"
+	"github.com/kinecosystem/go/services/horizon/internal/render/sse"
+	"github.com/kinecosystem/go/services/horizon/internal/test"
+)
+
+// Check subscription gets updates, and sse.Tick() doesnt trigger channel.
+func TestSSEPubsub(t *testing.T) {
+	ht := StartHTTPTest(t, "base")
+	defer ht.Finish()
+
+	ht.App.ticks.Stop()
+
+	subscription := sse.Subscribe("a")
+	defer sse.Unsubscribe(subscription, "a")
+
+	sse.Publish("a")
+
+	select {
+	case <-subscription:
+		// no-op.  Success!
+	case <-time.After(2 * time.Second):
+		t.Error("channel did not trigger")
+		t.FailNow()
```

Comment:
doesn't `FailNow()` accept a string as well? if so, no need for `t.Error()` before that. this is repeated in other places in the pr